src/shapes/polygon.class.js

```diff
@@ -79,11 +79,11 @@
           maxX = max(points, 'x'),
           maxY = max(points, 'y');
 
-      this.width = (maxX - minX) || 1;
-      this.height = (maxY - minY) || 1;
+      this.width = (maxX - minX) || 0;
+      this.height = (maxY - minY) || 0;
 
-      this.minX = minX,
-      this.minY = minY;
+      this.minX = minX || 0,
```

Comment:
Accepting no points in polygon or polyline means that we have to have a fall back in case top and left are not specified.